public/controller/data_functions.php

```diff
@@ -135,44 +144,134 @@ function upload_homework($username, $semester, $course, $assignment_id, $homewor
     if (!is_open_assignment($class_config, $assignment_id)) {
         return array("error"=>"", "message"=>$assignment_id." is not a valid assignment");
     }
+    if ($num_parts !== get_num_parts($class_config, $assignment_id)) {
+        return array("error"=>"", "message"=>"Number of parts for this homework does not match configuration.  ".$num_parts." != ".get_num_parts($class_config, $assignment_id));
+    }
     $assignment_config = get_assignment_config($semester, $course, $assignment_id);
     if (!can_edit_assignment($username, $semester, $course, $assignment_id, $assignment_config)) {//Made sure the user can upload to this homework
         return array("error"=>"assignment_closed", "message"=>$assignment_id." is closed.");
     }
     //VALIDATE HOMEWORK CAN BE UPLOADED HERE
     //ex: homework number, due date, late days
 
-    $max_size = 50;
-    if (isset($assignment_config["max_submission_size"])) {
-        $max_size = $assignment_config["max_submission_size"];
-    }
+    // check if files from previous submission exists
+    if($svn_checkout == false) {
+      if(isset($previous_files)) {
 
+        // check if folder for this assignment exists
+        $assignment_path = $path_front_course."/submissions/".$assignment_id;
+        if(!file_exists($assignment_path)) {
+          display_error("Folder ".$assignment_path." from previous submission does not exist.");
```

Comment:
This error is rather meaningless to the user and displays too much info to the user about folder structure. Message "Folder does not exist for this assignment. Contact administrator to resolve this issue."